### ISSUE TITLE
inheritance and reset

### ISSUE THREAD

**Issue by alexanderlerch** [medioqrity/2022-MUSI6106#20]
https://github.com/medioqrity/2022-MUSI6106/blob/8e64c4aa89c7dfa6cfaf4047fbe654ba7f616ca3/src/FastConv/Convolver.cpp#L5

is it intentional that you call reset here but you also override reset in the inherited class? Seems kinda messy

**Comment by med1844** [comment]
yes it's intentional. Reset and init are both virtual methods that takes care of the class's own variables.
